test/Microsoft.AspNet.Mvc.FunctionalTests/ModelBindingTest.cs

```diff
@@ -1949,9 +1950,11 @@ public class ModelBindingTest
 
             // Assert
             var result = await ReadValue<UserWithAddress>(response);
+
+            // Reviewers: Slightly odd behavior is specific to unusual error cases. Happens because
+            // MutableObjectModelBinder no longer incorrectly creates a model when value providers have no data.
             var address = Assert.Single(result.Addresses);
```

Comment:
no.  `result` is non-`null` because the `UserWithAddress` parameter was bound in the controller.  however `PersonAddress` is a non-top-level POCO object that was previously bound even though no data existed for it.

sequence is something like:
- `MutableObjectModelBinder` sees data for its `Addresses` property exists
- `ComplexObjectDtoModelBinder` attempts to bind `Addresses`
- `CollectionModelBinder` starts with the `0` property because `bindingContext.ValueProvider.ContainsPrefixAsync()` returns `true` (a case it can normally handle)
- attempt to convert the raw value to a collection of `PersonAddress` fails so entry gets `null` value